tools/protodoc/protodoc.py

```diff
@@ -713,7 +713,14 @@ def Main():
   response = plugin_pb2.CodeGeneratorResponse()
   cprofile_enabled = os.getenv('CPROFILE_ENABLED')
 
-  for proto_file in request.proto_file:
+  for file_to_generate in request.file_to_generate:
```

Comment:
Why is this change needed? Presumably this is so that the output maps to what bazel query expects? Can you add some more comments?

Reply:
Basically, it has to do with the fact that we're invoked under a Bazel aspect, which is a traversal of a shadow build graph. You only want to generate for the current node, not any of the dependencies, otherwise you generate non-canonical versions of files and have O(n^2) rather than O(n) behavior.